internal/wasm/jit/asm/amd64/impl.go

```diff
@@ -0,0 +1,401 @@
+}
+
+// AssignSourceConstant implements asm.Node.AssignSourceConstant.
+func (n *nodeImpl) AssignSourceConstant(value int64) {
+	n.srcConst = value
+}
+
+// OffsetInBinary implements asm.Node.OffsetInBinary.
+func (n *nodeImpl) OffsetInBinary() int64 {
+	return n.offsetInBinary
+}
+
+// String implements fmt.Stringer.
+func (n *nodeImpl) String() (ret string) {
+	instName := instructionName(n.instruction)
+	switch n.types {
+	case operandTypesNoneToNone:
+		ret = instName
+	case operandTypesNoneToRegister:
+		ret = fmt.Sprintf("%s , %s", instName, registerName(n.dstReg))
+	case operandTypesNoneToMemory:
+		if n.dstMemIndex != asm.NilRegister {
+			ret = fmt.Sprintf("%s , [%s + 0x%x + %s*0x%x]", instName,
+				registerName(n.dstReg), n.dstConst, registerName(n.dstMemIndex), n.dstMemScale)
+		} else {
+			ret = fmt.Sprintf("%s , [%s + 0x%x]", instName, registerName(n.dstReg), n.dstConst)
+		}
+	case operandTypesNoneToBranch:
+		ret = fmt.Sprintf("%s , {%v}", instName, n.jumpTarget)
+	case operandTypesRegisterToNone:
+		ret = fmt.Sprintf("%s %s, ", instName, registerName(n.srcReg))
+	case operandTypesRegisterToRegister:
+		ret = fmt.Sprintf("%s %s, %s", instName, registerName(n.srcReg), registerName(n.dstReg))
+	case operandTypesRegisterToMemory:
+		if n.dstMemIndex != asm.NilRegister {
+			ret = fmt.Sprintf("%s %s, [%s + 0x%x + %s*0x%x]", instName, registerName(n.srcReg),
+				registerName(n.dstReg), n.dstConst, registerName(n.dstMemIndex), n.dstMemScale)
+		} else {
+			ret = fmt.Sprintf("%s %s, [%s + 0x%x]", instName, registerName(n.srcReg), registerName(n.dstReg), n.dstConst)
+		}
+	case operandTypesRegisterToConst:
+		ret = fmt.Sprintf("%s %s, 0x%x", instName, registerName(n.srcReg), n.dstConst)
+	case operandTypesMemoryToRegister:
+		if n.srcMemIndex != asm.NilRegister {
+			ret = fmt.Sprintf("%s [%s + %d + %s*0x%x], %s", instName,
+				registerName(n.srcReg), n.srcConst, registerName(n.srcMemIndex), n.srcMemScale, registerName(n.dstReg))
+		} else {
+			ret = fmt.Sprintf("%s [%s + 0x%x], %s", instName, registerName(n.srcReg), n.srcConst, registerName(n.dstReg))
+		}
+	case operandTypesMemoryToConst:
+		if n.srcMemIndex != asm.NilRegister {
+			ret = fmt.Sprintf("%s [%s + %d + %s*0x%x], 0x%x", instName,
+				registerName(n.srcReg), n.srcConst, registerName(n.srcMemIndex), n.srcMemScale, n.dstConst)
+		} else {
+			ret = fmt.Sprintf("%s [%s + 0x%x], 0x%x", instName, registerName(n.srcReg), n.srcConst, n.dstConst)
+		}
+	case operandTypesConstToMemory:
+		if n.dstMemIndex != asm.NilRegister {
+			ret = fmt.Sprintf("%s 0x%x, [%s + 0x%x + %s*0x%x]", instName, n.srcConst,
+				registerName(n.dstReg), n.dstConst, registerName(n.dstMemIndex), n.dstMemScale)
+		} else {
+			ret = fmt.Sprintf("%s 0x%x, [%s + 0x%x]", instName, n.srcConst, registerName(n.dstReg), n.dstConst)
+		}
+	case operandTypesConstToRegister:
+		ret = fmt.Sprintf("%s 0x%x, %s", instName, n.srcConst, registerName(n.dstReg))
+	}
+	return
+}
+
+type operandType byte
```

Comment:
comment where this is from, if from somewhere (ex if we defined this ourselves for convenience there's no See link needed!)